var/spack/repos/builtin/packages/muparser/package.py

```diff
@@ -22,19 +22,25 @@ class Muparser(Package):
     # https://github.com/beltoforion/muparser/pull/46
     patch("auto_ptr.patch", when="@2.2.5")
 
-    depends_on("cmake@3.1.0:", when="@2.2.6:", type="build")
+    variant("samples", default=True, description="enable samples")
+    variant("openmp", default=True, description="enable OpenMP support")
+    variant(
+        "wide_char", default=False, description="enable wide character strings in place of ASCII"
+    )
+    variant("shared", default=True, description="enable shared libs")
 
-    # Cmake build since 2.2.6
-    @when("@2.2.6:")
-    def install(self, spec, prefix):
-        cmake_args = ["-DENABLE_SAMPLES=OFF", "-DENABLE_OPENMP=OFF", "-DBUILD_SHARED_LIBS=ON"]
+    depends_on("cmake@3.1.0:", when="@2.2.6:", type="build")
 
-        cmake_args.extend(std_cmake_args)
+    # Non-CMake build system is not supported by windows
+    conflicts("platform=windows", when="@2.2.5")
```

Comment:
(minor) I recommend making this a range like `@:2.2.5` (even though there are no earlier versions now, if they were ever added I think this would be too specific).